Null pointer exception when trying to deserialize into Either[]

Hi,

I am trying to deserialize a json string into an object of type Either[MyError, MyResult]

I am using jackson=module-scala 2.7.4, scala v 2.11

I am using the following:

``` scala
case class MyInput(
                  val1: String,
                  val2: String
                )

case class MyResult(val1: Boolean,
                          val2: String,
                          val3: String,
                          val4: String,
                          val4: String,
                          val5: String)

case class MyError(status: Int)

case class MyType(input: MyInput, expectation: Either[MyError, MyResult])

val input = Input("Hello", "World")
val ra = MyResult(true, "ae", "nm", "am", "nc", "de")
val ira = MyType(input, Right(ra))

JacksonMapper.writeValueAsString(ira)
```

This produces the following json:

``` json
{
    "input":{
            "val1":"Hello",
            "val2":"World"
        },
    "expectation":{
        "r":{
                "val1":true,
                "val2":"ae",
                "val3":"nm",
                "val4":"am",
                "val5":"nc",
                "val6":"de"
            }
    }
}
```

When trying to deserialize using:

``` scala
JacksonMapper.readValue(req.body, classOf[MyType])
```

I receive the following null pointer exception:

``` bash
Caused by: java.lang.NullPointerException: null
    at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeWithType(BeanDeserializerBase.java:1017) ~[jackson-databind-2.7.4.jar:2.7.4]
    at com.fasterxml.jackson.module.scala.deser.EitherDeserializer.deserializeValue(EitherDeserializer.scala:43) ~[jackson-module-scala_2.11-2.7.4.jar:na]
    at com.fasterxml.jackson.module.scala.deser.EitherDeserializer.deserializeEither(EitherDeserializer.scala:55) ~[jackson-module-scala_2.11-2.7.4.jar:na]
    at com.fasterxml.jackson.module.scala.deser.EitherDeserializer.deserialize(EitherDeserializer.scala:67) ~[jackson-module-scala_2.11-2.7.4.jar:na]
    at com.fasterxml.jackson.module.scala.deser.EitherDeserializer.deserialize(EitherDeserializer.scala:12) ~[jackson-module-scala_2.11-2.7.4.jar:na]
```

Can you help?

--- a/src/main/scala/com/fasterxml/jackson/module/scala/deser/EitherDeserializer.scala
+++ b/src/main/scala/com/fasterxml/jackson/module/scala/deser/EitherDeserializer.scala
@@ -22,7 +22,7 @@ private class EitherDeserializer(javaType: JavaType,
       val containedType = javaType.containedType(param)
 
       val paramDeserializer = Option( ctxt.findContextualValueDeserializer(containedType, property) )
-      val typeDeserializer = Option(property).map(p => BeanDeserializerFactory.instance.findPropertyTypeDeserializer(ctxt.getConfig, containedType, p.getMember) )
+      val typeDeserializer = Option(property).flatMap(p => Option(BeanDeserializerFactory.instance.findPropertyTypeDeserializer(ctxt.getConfig, containedType, p.getMember)) )
 
       ElementDeserializerConfig( paramDeserializer, typeDeserializer )
     }

--- a/src/test/scala/com/fasterxml/jackson/module/scala/deser/EitherDeserializerTest.scala
+++ b/src/test/scala/com/fasterxml/jackson/module/scala/deser/EitherDeserializerTest.scala
@@ -3,6 +3,7 @@ package com.fasterxml.jackson.module.scala.deser
 import com.fasterxml.jackson.annotation._
 import com.fasterxml.jackson.databind.JsonNode
 import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.deser.EitherJsonTest.EitherField
 import com.fasterxml.jackson.module.scala.deser.EitherJsonTest.{BaseHolder, Impl, PlainPojoObject}
 import org.junit.runner.RunWith
 import org.scalatest.junit.JUnitRunner
@@ -55,6 +56,10 @@ class EitherDeserializerTest extends DeserializerTest with EitherJsonTestSupport
   it should "deserialize a seq wrapped Either" in {
     deserialize[Seq[Either[String, String]]]("""[{"l":"left"}]""") shouldBe Seq(Left("left"))
   }
+
+  it should "deserialize class with a field with Either" in {
+    deserialize[EitherField]("""{"either":{"r":{"a":"1","b":null,"c":1}}}""") shouldBe EitherField(Right(PlainPojoObject("1", None, 1)))
+  }
 }
 
 
@@ -85,6 +90,8 @@ object EitherJsonTest {
     def base = _base
     def base_=(base:Either[Base, Base]) { _base = base }
   }
+
+  case class EitherField(either: Either[Int,PlainPojoObject])
 
   case class PlainPojoObject(a: String, b: Option[String], c: Long)
 }

